src/display/editor/editor.js

```diff
@@ -1319,9 +1392,139 @@ class AnnotationEditor {
     if (this.isResizable) {
       this.#createResizers();
       this.#resizersDiv.classList.remove("hidden");
+      bindEvents(this, this.div, ["keydown"]);
     }
   }
 
+  /**
+   * onkeydown callback.
+   * @param {KeyboardEvent} event
+   */
+  keydown(event) {
+    if (
+      !this.isResizable ||
+      event.target !== this.div ||
+      event.key !== "Enter"
+    ) {
+      return;
+    }
+    this._uiManager.setSelected(this);
+    this.#savedDimensions = {
+      savedX: this.x,
+      savedY: this.y,
+      savedWidth: this.width,
+      savedHeight: this.height,
+    };
+    const children =
+      this.#allResizerDivs || Array.from(this.#resizersDiv.children);
+    if (!this.#allResizerDivs) {
+      this.#allResizerDivs = children;
```

Comment:
It seems that https://github.com/mozilla/pdf.js/pull/17062#discussion_r1344243988 was a stupid idea on my part, since testing locally it appears that the correct resizer now isn't focused after rotating the document.

Please revert that again, sorry about the wasted time!